src/templates/antora/antora.converter.ts

```diff
@@ -79,19 +76,27 @@ const cleanup = (str: string): string => {
   return filters.reduce((acc, filter) => filter(acc), str);
 };
 
-const getNameFromFullName = (name: string): string =>
-  name.split('.').slice(-1).join('');
-
 const getFilePathFromFullName = (name: string): string => {
   const filename = name.toLowerCase() === 'tinymce' ? 'tinymce.root' : name.toLowerCase();
-  return basePath + filename + '.adoc';
+  return 'apis/' + filename + '.adoc';
 };
 
-const generateTypeXref = (type: string): string =>
-  type.includes('tinymce', 0) ? 'xref:' + getFilePathFromFullName(type) + '[' + getNameFromFullName(type) + ']' : type;
+const getFilePathFromFullNameLegacy = (name: string): string => {
+  const folder = name.split('.').slice(0, -1).join('.') + '/';
+  const filename = name.toLowerCase() === 'tinymce' ? 'tinymce/root_tinymce' : name.toLowerCase();
+  return 'api/' + folder + filename + '.adoc';
+};
 
-const generateDefinedByXref = (definedBy: string): string =>
-  'xref:' + getFilePathFromFullName(definedBy) + '[' + getNameFromFullName(definedBy) + ']';
+const getFilePath = (structure: string): Function =>
+  structure === 'legacy' ? getFilePathFromFullNameLegacy : getFilePathFromFullName;
+
+const generateXref = (name: string, structure: string, title?: string): string => {
+  title = title || name.split('.').slice(-1).join('');
```

Comment:
This is variable mutation and it's not something we like to do at Tiny. So this should declare a new variable, though as noted in my previous comment `title` is never passed so we could just remove it instead so it becomes:
```ts
const title = name.split('.').slice(-1).join('');
```